backend/static/js/qleverUI.js

```diff
@@ -273,44 +279,53 @@ $(document).ready(function () {
   
   // Generating the various links for sharing.
   $("#sharebtn").click(async function () {
-    // Rewrite the query, normalize it, and escape quotes.
-    //
-    // TODO: The escaping of the quotes is simplistic and currently fails when
-    // the query already contains some escaping itself.
-    const queryRewritten = await rewriteQuery(
-      editor.getValue(), {"name_service": "if_checked"});
-    const queryRewrittenAndNormalizedAndWithEscapedQuotes =
-      normalizeQuery(queryRewritten).replace(/"/g, "\\\"");
-
-    // POST request to Django, for the query hash.
-    $.post('/api/share', { "content": queryRewritten }, function (result) {
-      log('Generating links for sharing ...', 'other');
-      var baseLocation = window.location.origin + window.location.pathname.split('/').slice(0, 2).join('/') + '/';
-
-      // The default media type for the curl command line link is TSV, but for
-      // CONSTRUCT queries use Turtle.
-      var mediaType = "text/tab-separated-values";
-      var apiCallCommandLineLabel = "Command line for TSV export (using curl)";
-      if (queryRewrittenAndNormalizedAndWithEscapedQuotes.match(/CONSTRUCT \{/)) {
-        mediaType = "text/turtle";
-        apiCallCommandLineLabel = apiCallCommandLineLabel.replace(/TSV/, "Turtle");
+    try {
+      // Rewrite the query, normalize it, and escape quotes.
+      //
+      // TODO: The escaping of the quotes is simplistic and currently fails when
+      // the query already contains some escaping itself.
+      const queryRewritten = await rewriteQuery(
+        editor.getValue(), {"name_service": "if_checked"});
+      const queryRewrittenAndNormalizedAndWithEscapedQuotes =
+        normalizeQuery(queryRewritten).replace(/"/g, "\\\"");
+      
+      if (editor.state.completionActive) { editor.state.completionActive.close(); }
+
+      // POST request to Django, for the query hash.
+      const response = await fetch("/api/share", {
+        method: "POST",
+        body: new URLSearchParams({ "content": queryRewritten })
+      });
+      if (response.ok) {
+        const result = await response.json();
```

Comment:
Similar question as above, why not fetch, then, ...?

However, we do it, it would be good to do it consistently.